core/perf_test/PerfTest_ViewCopy.hpp

```diff
@@ -42,220 +42,134 @@
 //@HEADER
 */
 
+#ifndef KOKKOS_CORE_PERFTEST_BENCHMARK_VIEW_COPY_HPP
+#define KOKKOS_CORE_PERFTEST_BENCHMARK_VIEW_COPY_HPP
+
 #include <Kokkos_Core.hpp>
-#include <gtest/gtest.h>
-#include <cstdio>
-#include <PerfTest_Category.hpp>
+#include <benchmark/benchmark.h>
+#include <cmath>
 
 namespace Test {
 
+void report_results(benchmark::State& state, double time);
+
 template <class ViewTypeA, class ViewTypeB>
-double deepcopy_view(ViewTypeA& a, ViewTypeB& b, int repeat) {
-  Kokkos::Timer timer;
-  for (int i = 0; i < repeat; i++) {
+void deepcopy_view(ViewTypeA& a, ViewTypeB& b, benchmark::State& state) {
+  for (auto _ : state) {
+    Kokkos::Timer timer;
     Kokkos::deep_copy(a, b);
+    Kokkos::fence();
```

Comment:
Why are you fencing inside the loop now? Also, note that the two-argument deep_copy implies a global fence before and after the copy anyway.

Reply:
Previously we were controlling iterations manually with the `for (int i = 0; i < repeat; i++)` loop and calculating the average result with `time1 = deepcopy_view(a, b, R) / R;`.
Running a single kernel per iteration seems to make sense since now `benchmark` is responsible for both iterations and calculating the statistics.